### PR TITLE
Fix inconsistent handling of empty index ranges and edge cases

### PR DESCRIPTION
**Pull request by uklotzde**
While fixing another bug I noticed some inconsistencies when handling empty index ranges and when calculating the _intersection_ or _span_ if one or both of the arguments are empty (= single point).

Those edge cases doesn't seem to occur for any of our use cases, but nevertheless fixing them is recommended. New tests cover those edge cases now. Exhaustive testing is inevitable here!!

Also affects 2.2, but I'm afraid to touch foundational code. Better rely on extensive alpha/beta testing.

### CODE REVIEW

File: src/util/indexrange.cpp [hunk 1]
```diff
@@ -55,51 +55,83 @@ IndexRange reverse(IndexRange arg) {
...
             }
         } else {
-            DEBUG_ASSERT(!"Cannot intersect index ranges with contrary orientations");
+            DEBUG_ASSERT(!"Cannot intersect index ranges with different orientations");
+            return IndexRange();
+        }
+    } else {
+        DEBUG_ASSERT(lhs.start() == lhs.end());
+        if (rhs.start() <= rhs.end()) {
+            const SINT start = std::max(lhs.start(), rhs.start());
```
Comment (daschuer) [inline src/util/indexrange.cpp]: lhs is a single point. The code returns this point if it is within the range of rhs. 
I would make this explicit to have a better readable code. 
if (rhs.start() <= lhs.start() && rhs.end() >= lhs.end()) {
    return lhs;
} else {
    return IndexRange();
}

**Comment by daschuer** [comment]
Thanks. LGTM.
